### PR TITLE
Update TSDB, PostGIS. Allow updates on existing DB.

### PR DESCRIPTION
**Pull request by jdk24**
Update Timescaledb 1.7.4 -> 2.1.0
Update PostGIS 3.0.2 -> 3.1.1

Verified data being brought over from existing stack volume after replacing images.

### CODE REVIEW

**Comment by chrismayer** [comment]
Are any actions needed after pulling this? Removing the DB Volume or similar or is this compatible to previously existing DB?

**Comment by jdk24** [comment]
@chrismayer Good point. The commands are

`ALTER EXTENSION timescaledb UPDATE; ALTER EXTENSION postgis UPDATE; `

and should be executed only on extension upgrades, not with the incremental updates on each startup as a timescaleDB upgrade might interfere with its DB policies.

**Review by chrismayer — COMMENTED** [review]
LGTM, thank you @jdk! I'd suggest to add the following (non-functional) additions:

  - Add a comment to the `docker-entrypoint.sh` where the original file it is taken from / based on
  - Add a short comment in the file at the code spot where you changed the original  `docker-entrypoint.sh` and for what purpose
  - Add some new lines at all files where missing (as suggested from github editor)

File: db/update-files.d/1_alter_raster_metadata.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+\c sauber_data
+
+CREATE TABLE image_mosaics.raster_metadata (
```
Comment (chrismayer) [inline db/update-files.d/1_alter_raster_metadata.sql:3]: Just for clarification: The `raster_metadata` table is no more created in the base SQL script?

Comment (jdk24) [reply db/update-files.d/1_alter_raster_metadata.sql]: Good point. 
Was originally: DROP TABLE raster_metadata, then recreate. 
Now: CREATE **IF NOT EXISTS** to make sure it does before adding a column.
Changed in newest commit.

**Review by chrismayer — APPROVED** [review]
LGTM, thank you very much for your ongoing effort on this.

I tested it locally and the DB fires up as expected.

**Pull request merged**